packages/ra-ui-materialui/src/list/filter/FilterListItem.spec.tsx

```diff
@@ -140,6 +147,44 @@ describe('<FilterListItem/>', () => {
                 <FilterListItem label="Foo" value={{ foo: 'bar' }} />
             </ListContextProvider>
         );
-        expect(getByText('Foo').parentElement?.dataset.selected).toBe('true');
+        expect(screen.getByText('Foo').parentElement?.dataset.selected).toBe(
+            'true'
+        );
+    });
+
+    it('should allow to customize isSelected and toggleFilter', () => {
+        const { container } = render(<Cumulative />);
+
+        expect(
+            Array.from(
+                container.querySelectorAll<HTMLElement>(
+                    '[data-selected="true"]'
+                )
+            ).map(item => item.textContent)
+        ).toEqual(['News', 'Tutorials']);
+
+        screen.getByText(/"category":.*\[.*"tutorials",.*"news".*\]/);
+
+        screen.getByText('News').click();
+
+        expect(
+            Array.from(
+                container.querySelectorAll<HTMLElement>(
+                    '[data-selected="true"]'
+                )
+            ).map(item => item.textContent)
+        ).toEqual(['Tutorials']);
+        screen.getByText(/"category":.*\[.*"tutorials".*\]/);
+
+        screen.getByText('Tutorials').click();
+
+        expect(
+            Array.from(
+                container.querySelectorAll<HTMLElement>(
+                    '[data-selected="true"]'
+                )
+            ).map(item => item.textContent)
+        ).toEqual([]);
+        screen.getByText(/{}/);
```

Comment:
Your test only checks for removal cases, you should also add one addition case IMO

Reply:
Added